spyder_kernels/customize/namespace_manager.py

```diff
@@ -39,49 +37,52 @@ class NamespaceManager:
     current_namespace is True, or a new namespace.
     """
 
-    def __init__(self, filename, namespace=None, current_namespace=False,
-                 file_code=None, stack_depth=1):
+    def __init__(
+        self,
+        shell,
+        filename,
+        current_namespace=False,
+        file_code=None,
+        context_locals=None,
+        context_globals=None,
+    ):
+        self.shell = shell
         self.filename = filename
-        self.ns_globals = namespace
```

Comment:
namespace is now passed as context_globals, so there is two code paths instead of 3